cypress/integration/collections_spec.js

```diff
@@ -0,0 +1,83 @@
+      cy.visit(host);
+
+      cy.get('nav li a').contains('Collections').should('exist').as('collections');
+      cy.get('@collections').should('have.attr', 'href', '#/collections');
+      cy.get('@collections').click();
+
+      cy.url().should('include', 'collections');
+      cy.get('.heading--xlarge').should('have.text', 'Collections');
+
+      cy.get('table tbody tr').its('length').should('be.gt', 1);
+    });
+
+    it('collections page displays a button to add a new collection', () => {
+      cy.visit(`${host}/#collections`);
+
+      cy.get('.heading--large').should('have.text', 'Collection Overview');
+      cy.get('a').contains('Add a Collection').should('exist').as('addCollection');
+      cy.get('@addCollection').should('have.attr', 'href', '#/collections/add');
+      cy.get('@addCollection').click();
+
+      // fill the form and submit
+      // we need to use fake collection under test/fake-api-fixtures/collections, so the
+      // verification would work
+      const collection = '{{}"name":"MOD09GQ","version":"006","dataType":"MOD09GQ"}';
+      cy.get('textarea').type(collection, {force: true});
+      cy.get('form').get('input').contains('Submit').click();
+
+      // displays the new collection
+      cy.get('.heading--xlarge').should('have.text', 'Collections');
+      cy.get('.heading--large').should('have.text', 'MOD09GQ / 006');
+      cy.url().should('include', '#/collections/collection/MOD09GQ/006');
+    });
+
+    it('collection page has buttons to edit or delete the collection', () => {
+      cy.visit(`${host}/#/collections/collection/MOD09GQ/006`);
+      cy.get('a').contains('Edit').should('exist').as('editCollection');
+      cy.get('@editCollection').should('have.attr', 'href').and('include', '#/collections/edit/MOD09GQ/006');
+      cy.get('@editCollection').click();
+
+      cy.get('.heading--large').should('have.text', 'Edit MOD09GQ___006');
+
+      // cy.clear doesn't work for textarea, maybe it's related to https://github.com/cypress-io/cypress/issues/2056
+      // const collection = '{{}"name":"MOD09GQ","version":"006","dataType":"MOD09GQ"}';
+      // cy.get('textarea').clear({force: true}).type(collection, {force: true});
+      cy.get('textarea').type(' ', {force: true});
+      cy.get('form').get('input').contains('Submit').click();
+
+      // displays the updated collection and its granules
+      cy.get('.heading--xlarge').should('have.text', 'Collections');
+      cy.get('.heading--large').should('have.text', 'MOD09GQ / 006');
+      cy.get('table tbody tr').its('length').should('be.gt', 1);
+
+      // delete collection
+      cy.get('button').contains('Delete').should('exist').click();
+      cy.get('button').contains('Confirm').click();
+
+      // goes back to collections list
+      cy.url().should('include', 'collections');
+      cy.get('.heading--xlarge').should('have.text', 'Collections');
+      cy.get('table tbody tr').its('length').should('be.gt', 1);
```

Comment:
The length of the collections list was greater than 1 before this collection was deleted, correct? So I don't think asserting that the table length is still greater than 1 really proves that this specific collection was removed from the list. Perhaps we could do a `should('not.exist')` on an element in the table we expect to exist for this collection?

Reply:
With current fake-api server, the data isn't changed after `delete` request

Reply:
Ah, I see your point. One thing: if you delete the collection using the button from the list page instead of the detail page, the row does get removed and you can test against that. I'm doing this in #499.

If you want to stick with this approach though, we could get around the problem with Cypress stubbing for `cy.route`. This probably deserves a bigger conversation with @kkelly51 about how long we intend to keep the fake API and how we want to deal with faking dynamic data updates